### PR TITLE
{Package} Bump up azure-core to 1.31.0

### PR DESCRIPTION
**Pull request by kairu-ms**
**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Package

File: src/azure-cli/requirements.py3.Darwin.txt [hunk 1]
```diff
@@ -8,7 +8,7 @@ azure-cli-core==2.64.0
 azure-cli-telemetry==1.1.0
 azure-cli==2.64.0
 azure-common==1.1.22
-azure-core==1.28.0
+azure-core==1.30.0
```
Comment (kairu-ms) [inline src/azure-cli/requirements.py3.Darwin.txt:11]:
```suggestion
azure-core==1.31.0
```

File: src/azure-cli/requirements.py3.Linux.txt [hunk 2]
```diff
@@ -8,7 +8,7 @@ azure-cli-core==2.64.0
 azure-cli-telemetry==1.1.0
 azure-cli==2.64.0
 azure-common==1.1.22
-azure-core==1.28.0
+azure-core==1.30.0
```
Comment (kairu-ms) [inline src/azure-cli/requirements.py3.Linux.txt:11]:
```suggestion
azure-core==1.31.0
```

File: src/azure-cli/requirements.py3.windows.txt [hunk 3]
```diff
@@ -8,7 +8,7 @@ azure-cli-core==2.64.0
 azure-cli-telemetry==1.1.0
 azure-cli==2.64.0
 azure-common==1.1.22
-azure-core==1.28.0
+azure-core==1.30.0
```
Comment (kairu-ms) [inline src/azure-cli/requirements.py3.windows.txt:11]:
```suggestion
azure-core==1.31.0
```

**Comment by jiasli** [comment]
1.30.0 is not enough. `azure-identity` requires `azure-core>=1.31.0` since https://github.com/Azure/azure-sdk-for-python/pull/36882.

**Comment by jiasli** [comment]
Context: `azure-core==1.28.0` is not compatible with the latest `azure-identity` 1.18.0 which requires `azure-core>=1.31.0`  (https://github.com/Azure/azure-sdk-for-python/pull/36882). We have to bump `azure-core` so that the latest `azure-identity` can be installed in the same virtual environment.

**Review by jiasli — APPROVED** [review]

**Review by bebound — APPROVED** [review]

**Pull request merged**